pkg/pillar/zfs/zfs.go

```diff
@@ -60,6 +60,41 @@ func CreateDataset(datasetName string) error {
 	return MountDataset(datasetName)
 }
 
+// CreateSnapshot creates a snapshot of the dataset
+// Returns the name of the created snapshot that can be used to rollback
+func CreateSnapshot(datasetName string) (snapshotName string, err error) {
+	snapshotName = time.Now().Format("20060102-150405")
+	// add milliseconds to snapshot name
+	snapshotName += fmt.Sprintf("%03d", time.Now().Nanosecond()/1000000)
```

Comment:
You could use time.Millisecond instead of 1000000.

Reply:
Will do, thanks!